src/analysis/processing/qgsalgorithmrandompointsinpolygons.cpp

```diff
@@ -0,0 +1,394 @@
+                      "(<code>POLYGONS_WITH_MISSED_POINTS</code>).</li> "
+                      "<li> The number of features with an empty or no geometry "
+                      "(<code>LINES_WITH_EMPTY_OR_NO_GEOMETRY</code>).</li> "
+                      "</ul>"
+                    );
+}
+
+
+QgsRandomPointsInPolygonsAlgorithm *QgsRandomPointsInPolygonsAlgorithm::createInstance() const
+{
+  return new QgsRandomPointsInPolygonsAlgorithm();
+}
+
+bool QgsRandomPointsInPolygonsAlgorithm::prepareAlgorithm( const QVariantMap &parameters, QgsProcessingContext &context, QgsProcessingFeedback * )
+{
+  mNumPoints = parameterAsInt( parameters, POINTS_NUMBER, context );
+  mDynamicNumPoints = QgsProcessingParameters::isDynamic( parameters, POINTS_NUMBER );
+  if ( mDynamicNumPoints )
+    mNumPointsProperty = parameters.value( POINTS_NUMBER ).value< QgsProperty >();
+
+  mMinDistance = parameterAsDouble( parameters, MIN_DISTANCE, context );
+  mDynamicMinDistance = QgsProcessingParameters::isDynamic( parameters, MIN_DISTANCE );
+  if ( mDynamicMinDistance )
+    mMinDistanceProperty = parameters.value( MIN_DISTANCE ).value< QgsProperty >();
+
+  mMaxAttempts = parameterAsInt( parameters, MAX_TRIES_PER_POINT, context );
+  mDynamicMaxAttempts = QgsProcessingParameters::isDynamic( parameters, MAX_TRIES_PER_POINT );
+  if ( mDynamicMaxAttempts )
+    mMaxAttemptsProperty = parameters.value( MAX_TRIES_PER_POINT ).value< QgsProperty >();
+
+  mMinDistanceGlobal = parameterAsDouble( parameters, MIN_DISTANCE_GLOBAL, context );
+
+  mUseRandomSeed = parameters.value( SEED ).isValid();
+  mRandSeed = parameterAsInt( parameters, SEED, context );
+  mIncludePolygonAttr = parameterAsBoolean( parameters, INCLUDE_POLYGON_ATTRIBUTES, context );
+  return true;
+}
+
+QVariantMap QgsRandomPointsInPolygonsAlgorithm::processAlgorithm( const QVariantMap &parameters,
+    QgsProcessingContext &context, QgsProcessingFeedback *feedback )
+{
+  //::mTotNPoints = 0;
+  std::unique_ptr< QgsProcessingFeatureSource > polygonSource( parameterAsSource( parameters, INPUT, context ) );
+  if ( !polygonSource )
+    throw QgsProcessingException( invalidSourceError( parameters, INPUT ) );
+
+  QgsFields fields;
+  fields.append( QgsField( QStringLiteral( "rand_point_id" ), QVariant::LongLong ) );
+  if ( mIncludePolygonAttr )
+    fields.extend( polygonSource->fields() );
+
+  QString ldest;
+  std::unique_ptr< QgsFeatureSink > sink( parameterAsSink( parameters, OUTPUT,
+                                          context, ldest, fields, QgsWkbTypes::Point, polygonSource->sourceCrs() ) );
+  if ( !sink )
+    throw QgsProcessingException( invalidSinkError( parameters, OUTPUT ) );
+
+  QgsExpressionContext expressionContext = createExpressionContext( parameters, context, polygonSource.get() );
+
+  // Initialize random engine
+  std::random_device rd;
+  std::mt19937 mt( !mUseRandomSeed ? rd() : mRandSeed );
+  std::uniform_real_distribution<> uniformDist( 0, 1 );
```

Comment:
I don't think we require this -- if an explicit seed isn't set, then we just pass 0 as the seed to randomPointsInPolygon and let it take control of the random selection. If a seed is set, we pass that to randomPointsInPolygon directly